src/bls12_381/ec/g1.rs

```diff
@@ -35,6 +36,46 @@ impl fmt::Debug for G1Uncompressed {
     }
 }
 
+/// These methods provide fast reading and writing for `G1Affine` points.
+/// Points are guaranteed to be unaffected by a `write`-`read` roundtrip,
+/// but input to `read` is assumed to be correct. No validation is performed
+/// on the raw components, so it is an error to `read` arbitrary data.
+impl G1Affine {
+    #[inline]
+    pub fn raw_fmt_size() -> usize {
+        let s = G1Uncompressed::size();
+        s + 1
+    }
+
+    pub fn write<W: Write>(&self, mut writer: W) -> Result<usize, std::io::Error> {
+        if self.infinity {
```

Comment:
why not implement the io::Write trait?

Reply:
To this and complementary question: because these methods don't implement that interface. I changed to `write_raw` and `read_raw` to make the difference clearer.